src/db/connection.cr

```diff
@@ -21,10 +21,12 @@ module DB
   abstract class Connection
     include Disposable
     include SessionMethods(Connection, Statement)
+    include BeginTransaction
 
     # :nodoc:
     getter database
     @statements_cache = StringKeyCache(Statement).new
+    @transaction = false
```

Comment:
Why not store the `TopLevelTransaction` object itself here, so we can keep a reference to it from the Connection?